marlowe-runtime/tx/Language/Marlowe/Runtime/Transaction/Server.hs

```diff
@@ -307,24 +315,28 @@ execCreate contractQueryConnector getCurrentScripts solveConstraints loadWalletC
       , payoutScriptHash = payoutScript
       }
   let
-    contractSafetyErrors =
-      if False  -- FIXME: Disabled because of incompatibility with integration tests.
-        then checkContract networkId roleTokens version contract' continuations
-        else mempty
-  -- FIXME: The is a placeholder until we design safety-analysis reporting.
-  unless (null contractSafetyErrors)
-    . throwE . CreateSafetyAnalysisError
-    $ show contractSafetyErrors
+    -- Fast analysis of safety: examines bounds for transactions.
+    contractSafetyErrors = checkContract networkId roleTokens version contract' continuations
+    limitAnalysisTime =
+      liftIO
+        . fmap (either Right id)
+        . race ([SafetyAnalysisTimeout] <$ threadDelay (floor $ nominalDiffTimeToSeconds analysisTimeout * 1_000_000))
+  -- Slow analysis of safety: examines all possible transactions.
   transactionSafetyErrors <-
     ExceptT
       $ first CreateSafetyAnalysisError
-      <$> if False  -- FIXME: Disabled because of incompatibility with integration tests.
-            then checkTransactions solveConstraints version marloweContext rolesCurrency (changeAddress addresses) (toInteger minAda) contract' continuations
-            else pure $ pure mempty
-  -- FIXME: The is a placeholder until we design safety-analysis reporting.
-  unless (null transactionSafetyErrors)
-    . throwE . CreateSafetyAnalysisError
-    $ show transactionSafetyErrors
+      <$> limitAnalysisTime
+      (
+        checkTransactions
+          solveConstraints
+          version
+          marloweContext
+          rolesCurrency
+          (changeAddress addresses)
+          (toInteger minAda)
+          contract'
+          continuations
```

Comment:
Does this actually work? Aren't the safety errors lazily computed in `checkTransactions`? Z3 will run in IO, so that's fine, that will run in the eval thread, but at the end we have

```hs
      either throwE (pure . mconcat)
        . forM transactions
        $ checkTransaction solveConstraints version marloweContext rolesCurrency changeAddress
```

Which is a bit complex to reason about regarding laziness. Ultimately, the final `IO` action depends on pattern matching the `Either` returned by `forM`, and transitively from each call to `checkTransaction`, so that will force the evaluation of the fake `solveConstraints` calls.  The only unevaluated expressions here would be `pure . TransactionValidationError transaction . show` and `const $ TransactionWarning <$> V1.txOutWarnings txOutput`. How slow is `txOutWarnings`?

Reply:
The `txOutWarnings` is fast because it is computed from the semantics, not from executing Plutus. However, I think we should demonstrate this as a test case, so I've created the separate ticket PLT-6362.